api/provider/aws/deploy_create.go

```diff
@@ -0,0 +1,102 @@
+package aws
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+func (d *DeployProvider) Create(req models.CreateDeployRequest) (*models.Deploy, error) {
+	// TODO: see Line 59. Which entity ID generation pattern do we prefer? ARN to ID or the
+	// example below?
+	deployID := generateEntityID(req.DeployName)
+
+	if err := req.Validate(); err != nil {
+		return nil, err
+	}
+
+	deployFile, err := d.createRenderedDeploy(req.DeployFile)
+	if err != nil {
+		return nil, err
+	}
+
+	familyName := L0DeployID(d.Config.Instance(), req.DeployName)
```

Comment:
ahh! don't reintroduce the `L0DeployID` garbage!
The new pattern is something like:
```
deployID := generateEntityID(req.DeployName)
fqDeployID := addLayer0Prefix(deployID)
```

Reply:
✅ there was some edge case (or what looked like it) handling of quotes and prefix mangling happening here; still trying to keep track of the reasons for the 3-4 extra helper functions in `id.go` before I delete them.